Assets/scripts/UnitManager.cs

```diff
@@ -186,4 +188,20 @@ void Start()
 		}
 
 	}
+
+	void OnExitBattle(ExitBattleMessage m) {
```

Comment:
@GregoryElliott Look at this chunk of code for why damage isn't being dealt. Try adding a breakpoint here and see what you get.